### DISCUSSION TITLE
Isn't there an all in one package to buy?

### DISCUSSION THREAD

**Discussion by Piranhazone** [aclap-dev/video-downloadhelper#1216] [Help and Q&A]
### Your question

What's the point of buying lifetime licenses for just one specific browser?
What happens when a browser gets a complete remake including name change, like Microsoft did when they moved from IE to Edge? Does that end the "lifetime"?
I wouldn't mind a higher price, but I would highly prefer a lifetime license independent of browser, but I'm guessing that doesn't exist.
One can buy all 3 licenses one by one, but a single purchase for all in one doesn't exist? Why not?

Last but not least, what about the CoApp? Does that one remain free or is it included in the browser license or does it require another license?

### Addon & CoApp version and details

version: 9.1.0.43
target: microsoft
channel: stable
lang: de
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36 Edg/128.0.0.0
{
    "download_directory": "E:\\Download",
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "open_count_store": 1,
    "successfull_dl": 2
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

You have correctly discerned that there are 3 separate licenses.  There is no 3-in-1 package.  I can't address the reasons behind the business model for VDH.  Perhaps @jcfrog Jérôme can do that.  I would surmise that if a browser just changes names, VDH for the browser by its original name would continue to work on the browser by its new name, & your license would remain in effect.  Again, I can't give you official company pronouncements on this one, but this would be my expectation.  Firefox hasn't changed names since Netscape, & VDH came along after that.  I don't foresee Mozilla bothering to change its name any time soon, nor Google changing the name Chrome.  I don't believe there was a VDH for IE.  I think Edge was enough of an improvement over IE that making VDH work on Edge when it already worked on Firefox & Chrome was a minimal effort.  But as just another user, I'm speculating on that, especially since I never used IE & don't use Edge.

As for why there's different licenses per browser, that's due to differences in features.  Most glaringly, VDH for Chrome is not permitted to download from YouTube.  The owner of both Chrome & YouTube is Google.  Their rules say that no Chrome extension available on the Chrome web store is permitted to download from YouTube.  I don't use Chrome so I'm not certain of all the ins & outs.  But I believe Google has their own Chrome extension for downloading from YouTube.  But like I say, I don't use that browser so I could be wrong.  In any case, the crippling of VDH for Chrome explains its lower price.  I don't use Edge either so I don't know what VDH for Edge can & can't do vs VDH for Firefox, which I do use.  I do know that on Firefox & Edge, VDH downloads quite happily from YouTube.  Google's monopoly influence can't touch those 2 browsers.  I also know that primary development for VDH is done on Firefox (I believe on Linux).  A beta version of VDH for Firefox is always available, & that's the one you should be using if you're on Firefox.  The other 2 browsers don't get beta versions.  The bug corrections that go into the VDH for Firefox only eventually make their way into the general availability versions for the other 2 browsers.

As for the CoApp, there is no separate license for that.  It's the same CoApp for all the browsers.  You could say it's free.  You could say your license for the browser extension covers the CoApp.  Either way, anybody can get the CoApp & it doesn't cost anything extra.  The CoApp is different, though, across Windows vs Mac vs Linux.

You need to understand that it is not necessary to buy a license in order to use the browser extension.  It's the same program, with or without a license.  It has the same features & works 99.99% the same with & without a license.  You don't unlock any new features when you buy a license.  You don't get VDH to work on more web sites when you buy a license.  The primary thing you get is the absence of the annoying watermark on your downloaded videos.  The presence of the watermark is meant as an encouragement to buy a license.  In addition, on Chrome (and I believe Edge as well) there is a limit of 1 HLS download per 2-hour period when you don't have a license.  On Firefox, there is no 2-hour limit.  Licenses are available on Windows & Mac.  Linux is near & dear to the heart of the developer so there is no license on Linux.  You should not buy a license until you are certain that your favorite use cases actually work on VDH.  The web is extremely chaotic when it comes to multimedia content.  I find it somewhat miraculous that VDH does work on as many sites as it does.  But it does not claim to work on every site.  So you should test drive it before you buy it.

**Reply by Piranhazone** [reply]
Many thanks for this detailed answer.
The "limit of 1 HLS download per 2-hour period when you don't have a license" is exactly why I want the license, because I use VDH as little as twice a year or so, but when I do I usually want to download over a dozen videos all HLS.
At the same time using it so rarely makes it worth buying a license only if I can really be sure it's a lifetime license and not the lifetime of a browser that might reach the end of its life in a couple of years.
Firefox I've had a load of problems with, not only in browsing but just installing it without running it has caused me problems with other software I use, I rather don't install it anymore and either way, as I said, I don't mind paying if it's just a once in a lifetime.
Furthermore it has to be Edge because of the Google limitation where Edge is the one I suspect may get another remake/rename more likely than the others.

**Reply by GrampaWildWilly** [reply]
Lifetime really does mean as long as you are alive.  I would be very surprised if VDH did a bait & switch if Edge were renamed at some future date.  You never know what Microsoft has in store for us, but I think VDH will adapt.

**Comment by RustyBucket93** [comment]
Hi, I wonder if you could and sorry If I’m putting this in the wrong place. I want to buy a license but I’m using the Brave Brower. So which version of the license should I buy

**Reply by Mattmjs** [reply]
Brave is chromium so you likely have the extension from the chrome store, therefor you'd be getting the chrome license.
Keep in mind it won't work on YouTube, you need Firefox or Edge for that.
